File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/ServerNetworkInfo.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ServerNetworkInfo._

case class ServerNetworkInfo (
  /* List of VLANs. */
  vlans: List[String],
/* List of IPv6 VLANs. */
  vlans6: List[String],
assets: ServerNetworkInfoAssets,
switchports: ServerNetworkInfoSwitchports)

object ServerNetworkInfo {
  import DateTimeCodecs._

  implicit val ServerNetworkInfoCodecJson: CodecJson[ServerNetworkInfo] = CodecJson.derive[ServerNetworkInfo]
  implicit val ServerNetworkInfoDecoder: EntityDecoder[ServerNetworkInfo] = jsonOf[ServerNetworkInfo]
  implicit val ServerNetworkInfoEncoder: EntityEncoder[ServerNetworkInfo] = jsonEncoderOf[ServerNetworkInfo]
}

File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/ServerNetworkInfo.js
goog.provide('API.Client.ServerNetworkInfo');

/**
 * @record
 */
API.Client.ServerNetworkInfo = function() {}

/**
 * List of VLANs.
 * @type {!Array<!string>}
 * @export
 */
API.Client.ServerNetworkInfo.prototype.vlans;

/**
 * List of IPv6 VLANs.
 * @type {!Array<!string>}
 * @export
 */
API.Client.ServerNetworkInfo.prototype.vlans6;

/**
 * @type {!API.Client.ServerNetworkInfo_assets}
 * @export
 */
API.Client.ServerNetworkInfo.prototype.assets;

/**
 * @type {!API.Client.ServerNetworkInfo_switchports}
 * @export
 */
API.Client.ServerNetworkInfo.prototype.switchports;

